api_client/methods_api.py

```diff
@@ -0,0 +1,52 @@
+from urllib.parse import urljoin
+
+from requests import Session
+
+from core.logger import LOGGER
+from core.settings import BASE_API_URL, OK
+
+
+class ApiClient:
+    def __init__(self):
+        self.base_url = BASE_API_URL
+        self.session = Session()
+
+    @staticmethod
+    def log_pre(method, url, headers, data, expected_status):
+        LOGGER.info(f'Performing {method} request:\n'
+                    f'URL: {url}\n'
+                    f'HEADERS: {headers}\n'
+                    f'DATA: {data}\n\n'
+                    f'Expected status: {expected_status}\n\n')
+
+    @staticmethod
+    def log_post(response):
+        log_str = f'Got response:\n' \
+                  f'RESPONSE STATUS: {response.status_code}'
+        LOGGER.info(f'{log_str}\n'
+                    f'RESPONSE CONTENT: {response.text}\n\n')
+
+    def request(self, method, location, headers=None, data=None, expected_status=200):
```

Comment:
Looks like the expected status is always 200, and you cannot pass the custom expected status to this method

Reply:
No, any status can be passed in the request() method.
E.g. result = self.request('POST', LOGIN_API_URL, headers, expected_status=404)
I use this approach in post_login() method to test both positive and negative scenarios.